site/content/blog/package-authoring-with-kctrl.md

```diff
@@ -0,0 +1,56 @@
+---
+title: "Introducing kctrl package authoring commands"
+slug: kctrl-pkg-authoring-cmds
+date: 2022-08-19
+author: Rohit Aggarwal, Soumik Majumder
+excerpt: "Learn about the package authoring commands introduced in kctrl and how it makes it easier for package authors to create Carvel packages."
+image: /img/logo.svg
+asciinema: true
+tags: ['Rohit Aggarwal', 'Soumik Majumder', 'kapp-controller', 'kctrl', 'carvel']
+---
+
+In today's post, we are going to see how the kctrl CLI eases the process of package authoring.
+
+A [package](https://carvel.dev/kapp-controller/docs/latest/packaging/#package) is a combination of configuration metadata and OCI images. It informs the package manager what software it holds and how to install itself onto a Kubernetes cluster.
+
+A package author encapsulates, versions and distributes Kubernetes manifests as package for package consumers to install on a Kubernetes cluster. They can choose to create a package by using a third party manifest, e.g. they can choose to create a package from [cert-manager](https://cert-manager.io/), [Dynatrace](https://www.dynatrace.com/), etc., or they can distribute their own project Kubernetes manifest by creating a package.
+
+
+## Typical Package Authoring journey
+Package Authoring is an iterative process and below are the most common steps performed by the authors:
+1. Authors know about the Kubernetes manifest they want to package.
+2. Add/change the manifest by adding additional [overlay](https://carvel.dev/ytt/docs/latest/ytt-overlays/)/[template](https://carvel.dev/ytt/docs/latest/#templating) and test the package. This is the iterative part where authors want to make the changes and test them quickly.
+3. Once all the manifest are in place, create the [imgpkg](https://carvel.dev/imgpkg/) bundle (to be mentioned in the package) and the package itself.
+4. Add the package to the package repository for distribution.
+
+## How does `kctrl` help package authors?
+Today, package authors are supposed to know tools like [imgpkg](https://carvel.dev/imgpkg/), [kbld](https://carvel.dev/kbld/), [vendir](https://carvel.dev/vendir/), etc. before they start on the package authoring journey. These Carvel tools has a learning curve of itself. We wanted to introduce a set of CLI commands that guide users in performing most common packaging steps so that they are able to create and release the package without knowing these tools.
```

Comment:
```suggestion
Today, package authors are supposed to know about tools like [imgpkg](https://carvel.dev/imgpkg/), [kbld](https://carvel.dev/kbld/), [vendir](https://carvel.dev/vendir/), etc. before they start on their package authoring journey. These Carvel tools have a learning curve by themselves. We wanted to introduce a set of CLI commands that guide users while performing common packaging steps so that they are able to create and release the package without being familiar with these tools.
```

Reply:
Today, before package authors start on their authoring journey, they are required to know about tools like [imgpkg](https://carvel.dev/imgpkg/), [kbld](https://carvel.dev/kbld/) and [vendir](https://carvel.dev/vendir/), which are building blocks for Carvel packages. Our intention is to introduce a set of CLI commands that would guide users through the common packaging steps to enable them to create and release the packages without being familiar with these tools, while also learning about them in the process. We aim to achieve most of the scenarios using kctrl. For more complex use cases authors can always leverage these tools.

Reply:
@cppforlife thoughts?